src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -174,5 +174,92 @@ public void editMessage(String messageId, String messageText)
       System.out.println("error: " + e.toString());
     }
   }
+  
+  public void storeLocation(Location location)
+  {
+    Entity userEntity = new Entity("Location", location.getId().toString());
+    userEntity.setProperty("longitude", location.getLongitude()+"");
```

Comment:
Try using String.valueOf(double) here and below.

Reply:
Thank you, I will.